llvm/test/CodeGen/RISCV/intrinsic-cttz-elts.ll

```diff
@@ -12,9 +12,11 @@ define i16 @ctz_v4i32(<4 x i32> %a) {
 ; RV32-NEXT:    vsetvli zero, zero, e8, mf4, ta, ma
 ; RV32-NEXT:    vmv.v.i v8, 0
 ; RV32-NEXT:    vmerge.vim v8, v8, -1, v0
+; RV32-NEXT:    vsetivli zero, 1, e8, mf4, ta, ma
 ; RV32-NEXT:    vid.v v9
```

Comment:
This can be further optimized? since `vl=1` so there is only one element `0` in `v9`, thus `vrsub.vi v9, v9, 4` equals to `vmv.x.i v9, 4`.

Reply:
That is a nice find! I think this optimization should be done in a separate patch.

Reply:
Is this specific to VL though? I think this can hold for all VL when we know that all elements are 0 in the vector source?